src/coreclr/vm/callstubgenerator.cpp

```diff
@@ -0,0 +1,1011 @@
+    (PCODE)Load_D6,                         // 66
+    (PCODE)Load_D6_D7,                      // 67
+    (PCODE)0,                               // 70
+    (PCODE)0,                               // 71
+    (PCODE)0,                               // 72
+    (PCODE)0,                               // 73
+    (PCODE)0,                               // 74
+    (PCODE)0,                               // 75
+    (PCODE)0,                               // 76
+    (PCODE)Load_D7                          // 77
+};
+#endif // TARGET_ARM64
+
+PCODE GetGPRegRangeLoadRoutine(int r1, int r2)
+{
+    int index = r1 * NUM_ARGUMENT_REGISTERS + r2;
+    return GPRegsRoutines[index];
+}
+
+#ifndef UNIX_AMD64_ABI
+PCODE GetGPRegRefLoadRoutine(int r)
+{
+    return GPRegsRefRoutines[r];
+}
+#endif // UNIX_AMD64_ABI
+
+PCODE GetFPRegRangeLoadRoutine(int x1, int x2)
+{
+    int index = x1 * NUM_FLOAT_ARGUMENT_REGISTERS + x2;
+    return FPRegsRoutines[index];
+}
+
+extern "C" void CallJittedMethodRetVoid(PCODE *routines, int8_t*pArgs, int8_t*pRet, int totalStackSize);
+extern "C" void CallJittedMethodRetDouble(PCODE *routines, int8_t*pArgs, int8_t*pRet, int totalStackSize);
+extern "C" void CallJittedMethodRetI8(PCODE *routines, int8_t*pArgs, int8_t*pRet, int totalStackSize);
+extern "C" void CallJittedMethodRetBuff(PCODE *routines, int8_t*pArgs, int8_t*pRet, int totalStackSize);
+
+#ifdef UNIX_AMD64_ABI
+extern "C" void CallJittedMethodRetI8I8(PCODE *routines, int8_t*pArgs, int8_t*pRet, int totalStackSize);
+extern "C" void CallJittedMethodRetI8Double(PCODE *routines, int8_t*pArgs, int8_t*pRet, int totalStackSize);
+extern "C" void CallJittedMethodRetDoubleI8(PCODE *routines, int8_t*pArgs, int8_t*pRet, int totalStackSize);
+extern "C" void CallJittedMethodRetDoubleDouble(PCODE *routines, int8_t*pArgs, int8_t*pRet, int totalStackSize);
+#endif
+
+#ifdef TARGET_ARM64
+extern "C" void CallJittedMethodRet2I8(PCODE *routines, int8_t*pArgs, int8_t*pRet, int totalStackSize);
+extern "C" void CallJittedMethodRet2Double(PCODE *routines, int8_t*pArgs, int8_t*pRet, int totalStackSize);
+extern "C" void CallJittedMethodRet3Double(PCODE *routines, int8_t*pArgs, int8_t*pRet, int totalStackSize);
+extern "C" void CallJittedMethodRet4Double(PCODE *routines, int8_t*pArgs, int8_t*pRet, int totalStackSize);
+extern "C" void CallJittedMethodRetFloat(PCODE *routines, int8_t*pArgs, int8_t*pRet, int totalStackSize);
+extern "C" void CallJittedMethodRet2Float(PCODE *routines, int8_t*pArgs, int8_t*pRet, int totalStackSize);
+extern "C" void CallJittedMethodRet3Float(PCODE *routines, int8_t*pArgs, int8_t*pRet, int totalStackSize);
+extern "C" void CallJittedMethodRet4Float(PCODE *routines, int8_t*pArgs, int8_t*pRet, int totalStackSize);
+#endif // TARGET_ARM64
+
+// Generate the call stub for the given method.
+// The returned call stub header must be freed by the caller using FreeCallStub.
+CallStubHeader *CallStubGenerator::GenerateCallStub(MethodDesc *pMD)
+{
+    CONTRACTL
```

Comment:
This should stay as `STANDARD_VM_CONTRACT;` and switch to preemptive mode if necessary. We do not want to be doing heavy lifting in cooperative mode.

Reply:
Done